src/AutoRest.CSharp/Common/Output/Builders/CSProjBuilder.cs

```diff
@@ -0,0 +1,187 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Xml;
+
+namespace AutoRest.CSharp.Common.Output.Builders;
+
+internal class CSProjBuilder
+{
+    public CSProjBuilder()
+    {
+        ProjectReferences = new List<CSProjDependencyPackage>();
+        PackageReferences = new List<CSProjDependencyPackage>();
+        CompileIncludes = new List<CSProjCompileIncldue>();
+    }
+
+    public CSProjProperty<string>? Description { get; init; }
+
+    public CSProjProperty<string>? AssemblyTitle { get; init; }
+
+    public CSProjProperty<string>? Version { get; init; }
+
+    public CSProjProperty<string>? PackageTags { get; init; }
+
+    public CSProjProperty<string>? TargetFrameworks { get; init; }
+
+    public CSProjProperty<string>? TargetFramework { get; init; }
+
+    public CSProjProperty<bool>? IncludeOperationsSharedSource { get; init; }
+
+    public CSProjProperty<string>? LangVersion { get; init; }
+
+    public CSProjProperty<bool>? GenerateDocumentationFile { get; init; }
+
+    public CSProjProperty<string>? NoWarn { get; init; }
+
+    public IList<CSProjDependencyPackage> ProjectReferences { get; }
+
+    public IList<CSProjDependencyPackage> PackageReferences { get; }
+
+    public IList<CSProjCompileIncldue> CompileIncludes { get; }
+
+    public string Build()
+    {
+        var builder = new StringBuilder();
+        using var writer = XmlWriter.Create(builder, new XmlWriterSettings
+        {
+            OmitXmlDeclaration = true,
+            Indent = true
+        });
```

Comment:
I choose `XmlWriter` instead of constructing a tree of `XmlElement` or `XElement` (which is much more convenient)
because these `Element` stuff does not support we to add empty lines between those elements.
And we need to add blank lines in csproj files to make them more readable.
For the same reason, `Microsoft.Build` package is not useful here even if it could construct a csproj file in a way with semantics.

Reply:
Will this be used for production csproj file generation?
Or how important is it to keep the empty lines between elements? 
If we can leverage existing packages, it will reduce the code complexity significantly.

Reply:
It is not used in our SDK.
It might be used by external customers. It is true that this blank line has relatively lower priority.
But I would like to reduce the amount of changes.
There will be follow up PRs on the csproj related things, we could change it there.

Reply:
On the other hand, since XmlWriter provides more low-level API, it will have less allocation and better performance then we are using XmlElement or XElement or Microsoft.Build package.
It is also kind of straightforward enough to write the xml file.

Reply:
Not sure how this might be used for external customers since the class is `internal`.
And if this is used for external customers, I would prefer not to re-invent wheels to avoid maintenance effects.
I think we need to evaluate different ways by gathering more data, such as performance and allocation.
Given this is only used for testing internally, I am OK to merge it firstly and we revisit it with more data.

Reply:
No this is not the point.
The classes in our repo, are always only used by us.
The feature of generating csproj, is used by us only when generating test projects in this repo, and by external customers when they are generating their libraries.
This feature of generating csproj is controlled by config `skip-csproj`. It is always set to true when generating the libraries in `azure-sdk-for-net`.

We do not need to consider the possibility of "external customer needs more things so that they extend our generator to do that". That is not possible now